app/controllers/application_controller.rb

```diff
@@ -36,4 +36,11 @@ def destroy_unsaved_npcs
     end
     NpcModel.where(is_saved: 0).destroy_all
   end
+
+  def check_if_saving_npc
+    if NpcModel.where(is_saved: 0).count.positive?
+      UserNpcModel.create!(npc_model_id: NpcModel.where(is_saved: 0)[0].id, user_id: current_user.id)
+      NpcModel.where(is_saved: 0)[0].update(is_saved: 1)
+    end
+  end
```

Comment:
This was being used in both the Admin Dashboard and the User Dashboard, so I've made it a helper method